platform/ios/src/MGLAnnotationView.m

```diff
@@ -0,0 +1,72 @@
+#import "MGLAnnotationView.h"
+#import "MGLAnnotationView_Private.h"
+
+@interface MGLAnnotationView ()
+
+@property (nonatomic, weak) id<MGLAnnotationViewDelegate> delegate;
+@property (nonatomic, copy, nullable) NSString *reuseIdentifier;
+@property (nonatomic) id<MGLAnnotation> annotation;
+
+@end
+
+@implementation MGLAnnotationView
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self)
+    {
+        [self commonInit];
+    }
+    return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame
+{
+    self = [super initWithFrame:frame];
+    if (self)
+    {
+        [self commonInit];
+    }
+    return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)coder
+{
+    self = [super initWithCoder:coder];
+    if (self)
+    {
+        [self commonInit];
+    }
+    return self;
+}
+
+- (void)commonInit
+{
+    UITapGestureRecognizer *tapGestureRecognizer = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(didTapView:)];
+    [self addGestureRecognizer:tapGestureRecognizer];
```

Comment:
I suspect that having a separate gesture recognizer for each annotation would have a nontrivial impact on performance. We already have a gesture recognizer for the overall MGLMapView that calls `-handleSingleTapGesture:`. Can we call `-hitTest:` on each of the annotations there?